spring-grpc-core/src/test/java/org/springframework/grpc/client/ClientInterceptorsConfigurerTests.java

```diff
@@ -168,6 +169,35 @@ void whenBlendInterceptorsTrueThenGlobalInterceptorsBlended() {
 
 	}
 
+	@Nested
+	class WithInterceptorFilters {
+
+		@Test
+		void whenFilterExcludesOneGlobalInterceptor_thenBuilderGetsOnlyAllowedOnes() {
+			ManagedChannelBuilder<?> builder = Mockito.mock();
+			ClientInterceptorsConfigurerTests.this.contextRunner()
+				.withUserConfiguration(GlobalClientInterceptorsConfig.class, FilterConfig.class)
```

Comment:
This is fine, but you can get rid of the extra config class and a bean directly like (either is fine, up to you, just passing info along)...

```java
...
.withBean(ClientInterceptorFilter.class, () -> (interceptor, factory) -> interceptor == GlobalClientInterceptorsConfig.GLOBAL_INTERCEPTOR_BAR)
```